tests/functional/generate_pages.py

```diff
@@ -0,0 +1,190 @@
+import json
+import logging
+import os
+import sys
+import re
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+SPEC_PAGE_HEADER = r"""import chai from 'chai'
+import {startQuestionnaire} from '../helpers'
+
+"""
+
+SPEC_PAGE_IMPORT = r"""import {pageName} from '../pages/surveys/census/household/{pageFile}'
+"""
+
+SPEC_CHAI_HEADER = r"""
+const expect = chai.expect
+"""
+
+SPEC_EXAMPLE_TEST = r"""
+describe('Example Test', function() {
+
+  it('Given..., When..., Then...', function() {
+    startQuestionnaire('{schema}')
+  })
+
+})
+
+"""
+
+HEADER = r"""import {basePage} from '../../{basePageFile}'
+
+"""
+
+CLASS_NAME = r"""class {pageName}Page extends {basePage} {
+
+"""
+
+ANSWER_SETTER = r"""  set{answerName}(value) {
+    browser.setValue('[name="{answerId}"]', value)
+    return this
+  }
+
+"""
+
+ANSWER_GETTER = r"""  get{answerName}(value) {
+    return browser.element('[name="{answerId}"]').getValue()
+  }
+
+"""
+
+CHECKBOX_RADIO_CLICKER = r"""  click{optionName}() {
+    browser.element('[id="{optionId}"]').click()
+    return this
+  }
+
+"""
+
+FOOTER = r"""}
+
+export default new {pageName}Page()
+"""
+
+
+def generate_camel_case_from_id(id_str):
+    parts = re.sub('[^0-9a-zA-Z]+', '-', id_str).split('-')
+    name = ''.join([p.title() for p in parts])
+    return name
+
+
+def process_options(answer_id, options, page_spec):
+    for index, option in enumerate(options):
+        option_name = generate_camel_case_from_id(option['value'])
+        option_id = "{name}-{index}".format(name=answer_id, index=index+1)
+        page_spec.write(CHECKBOX_RADIO_CLICKER.replace("{optionName}", option_name).replace("{optionId}", option_id))
+
+
+def process_answer(answer, page_spec):
+    if answer['type'] == 'Radio' or answer['type'] == 'Checkbox':
+        process_options(answer['id'], answer['options'], page_spec)
+
+    if answer['type'] == 'Date':
+        answer_name = generate_camel_case_from_id(answer['id'])
+        page_spec.write(ANSWER_SETTER.replace("{answerName}", answer_name + 'Day').replace("{answerId}", answer['id'] + '-day'))
+        page_spec.write(ANSWER_GETTER.replace("{answerName}", answer_name + 'Day').replace("{answerId}", answer['id'] + '-day'))
+        page_spec.write(ANSWER_SETTER.replace("{answerName}", answer_name + 'Month').replace("{answerId}", answer['id'] + '-month'))
+        page_spec.write(ANSWER_GETTER.replace("{answerName}", answer_name + 'Month').replace("{answerId}", answer['id'] + '-month'))
+        page_spec.write(ANSWER_SETTER.replace("{answerName}", answer_name + 'Year').replace("{answerId}", answer['id'] + '-year'))
+        page_spec.write(ANSWER_GETTER.replace("{answerName}", answer_name + 'Year').replace("{answerId}", answer['id'] + '-year'))
+
+    else:
```

Comment:
Could this actually be `elif answer['type'] == 'TextField':` ? possible with a final `else` that throws an Exception so it's obvious when the type isn't handled.